libs/driver/in-memory/src/routing/provider.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  EnvironmentProviders,
+  inject,
+  InjectionToken,
+  makeEnvironmentProviders,
+} from '@angular/core';
+
+import {
+  DaffInMemoryRoutableObject,
+  DaffInMemoryRoutableObjects,
+} from './routable-object';
+
+/**
+ * A function that returns an array of entities with URL properties.
+ * Used to provide routable in-memory object information to the routing system.
+ */
+export type DaffInMemoryRoutableObjectsResolver = () => Array<{ url: string }>;
+
+/**
+ * Multi-injection token for routable object resolvers.
+ *
+ * Each in-memory driver (e.g., product, navigation) can register a resolver
+ * that provides its entities with URL properties. This token collects all
+ * such resolvers so they can be processed together to build the routing map.
+ *
+ * @docs-private
+ */
+const DAFF_INMEMORY_ROUTABLE_OBJECTS_RESOLVER = new InjectionToken<{ type: string; resolver: DaffInMemoryRoutableObjectsResolver }[]>(
+  'DAFF_INMEMORY_ROUTABLE_OBJECTS_RESOLVER', {
+    providedIn: 'root',
+    factory: () => [],
+  },
+);
+
+/**
+ * Internal injection token that builds and provides the complete map of routable objects.
+ *
+ * This token automatically collects all registered routable object resolvers and builds
+ * a Map where the key is the URL path and the value is an object containing both the URL
+ * and its associated type. This map is used by in-memory drivers to resolve URLs to their types.
+ *
+ * @docs-private
+ */
+export const DAFF_INMEMORY_ROUTABLE_OBJECTS = new InjectionToken<DaffInMemoryRoutableObjects>(
+  'DAFF_INMEMORY_ROUTABLE_OBJECTS',
+  {
+    providedIn: 'root',
+    factory: () => {
+      const resolvers = inject(DAFF_INMEMORY_ROUTABLE_OBJECTS_RESOLVER, { optional: true }) || [];
+      const map = new Map<string, DaffInMemoryRoutableObject>();
+
+      if (!Array.isArray(resolvers)) {
+        throw new Error('DAFF_INMEMORY_ROUTABLE_OBJECTS_RESOLVER must be an array');
+      }
+
+      resolvers.forEach((entry) => {
+        const entities = entry.resolver();
```

Comment:
I think you need to invoke it with `runInInjectionContext(entry.resolver)`. AFAIK the injection context does not automatically transfer like this.

Reply:
Perhaps. I believe since it's called within a factory, this may have always been the case, but I'd need to try it out on older versions to be sure.

Reply:
huh yeah, testing in a playground indicates that injection context survives in child functions.